src/MiniHallway.jl

```diff
@@ -0,0 +1,88 @@
+# Mini Hallway problem defined in http://cs.brown.edu/research/ai/pomdp/examples/mini-hall2.POMDP.
+# Original idea published in Littman, Cassandra and Kaelbling's ML-95 paper.
+
+# Basic parameters are:
+# discount: 0.950000
+# values: reward
+# states: 13
+# actions: 3
+# observations: 9
+# The rest is available at link
+
+
+struct Observation
+    no::Int
+end
+
+struct MiniHallway <: POMDP{Int, Int, Observation}
+    T::Array{SparseCat, 2}
+    R::Array{Float64}
+    O::Array{SparseCat{Array{Observation, 1}, Array{Float64, 1}}}
+    discount::Float64
+end
+
+function MiniHallway()
+    T = Array{SparseCat, 2}(undef, 13, 3)
+
+    T[1,1] = SparseCat((1), (1.)); T[1,2] = SparseCat((2), (1.)); T[1,3] = SparseCat((4), (1.))
```

Comment:
See comment above about putting this in the `transition` function, but, just for future reference, you could create this Matrix easily without having to write out all the indices with the syntax
```
SC = SparseCat
T = SC[SC(1, 1.) SC(3, 1.) SC(1, 1.);
SC(1, 1.) SC(3, 1.) SC(1, 1.);
# etc
]
```